clinica/iotools/converters/adni_to_bids/adni_modalities/adni_fmap.py

```diff
@@ -0,0 +1,335 @@
+from os import Path, PathLike
+from typing import List, Optional
+import pandas as pd
+
+def convert_adni_fmap(
+    source_dir: PathLike,
+    csv_dir: PathLike,
+    destination_dir: PathLike,
+    conversion_dir: PathLike,
+    subjects: Optional[List[str]] = None,
+    mod_to_update: bool = False,
+):
+    """Convert fMR images of ADNI into BIDS format.
+
+    Parameters
+    ----------
+    source_dir : PathLike
+        Path to the ADNI directory.
+
+    csv_dir : PathLike
+        Path to the clinical data directory.
+
+    destination_dir : PathLike
+        Path to the destination BIDS directory.
+
+    conversion_dir : PathLike
+        Path to the TSV files including the paths to original images.
+
+    subjects : List of str, optional
+        List of subjects.
+
+    mod_to_update : bool
+        If True, pre-existing images in the BIDS directory
+        will be erased and extracted again.
+    """
+
+    from os import path
+
+    import pandas as pd
+
+    from clinica.iotools.converters.adni_to_bids.adni_utils import paths_to_bids
+    from clinica.utils.stream import cprint
+
+    if not subjects:
+        adni_merge_path = path.join(csv_dir, "ADNIMERGE.csv")
+        adni_merge = pd.read_csv(adni_merge_path, delimiter='","')
+        adni_merge.columns = adni_merge.columns.str.strip('"')
+        subjects = list(adni_merge.PTID.unique())
+    
+    cprint(
+        f"Calculating paths of fMRI field maps (FMAPs). Output will be stored in {conversion_dir}."
+    )
+
+    images = compute_fmap_path(source_dir, csv_dir, subjects, conversion_dir)
+    
+    cprint("Paths of field maps found. Exporting images into BIDS ...")
+
+    paths_to_bids(images, destination_dir, "fmap", mod_to_update=mod_to_update)
+    rename_fmaps(destination_dir)
+
+    cprint(msg="Field maps conversion done.", lvl="debug")
+
+def compute_fmap_path(source_dir: Path, csv_dir: Path, subjs_list: list[str], conversion_dir: Path) -> pd.DataFrame:
+    """Compute the paths to fMR images.
+
+    Args:
+        source_dir: path to the ADNI directory
+        csv_dir: path to the clinical data directory
+        subjs_list: subjects list
+        conversion_dir: path to the TSV files including the paths to original images
+
+    Returns:
+        pandas Dataframe containing the path for each fmri
+    """
+    from os import path
+
+    import pandas as pd
+
+    from clinica.iotools.converters.adni_to_bids.adni_utils import (
+        find_image_path,
+        visits_to_timepoints,
+    )
+
+    fmap_col = [
+        "Subject_ID",
+        "VISCODE",
+        "Visit",
+        "Sequence",
+        "Scan_Date",
+        "Study_ID",
+        "Field_Strength",
+        "Series_ID",
+        "Image_ID",
+    ]
+    fmap_df = pd.DataFrame(columns=fmap_col)
+    fmap_dfs_list = []
+    adni_merge = pd.read_csv(csv_dir / "ADNIMERGE.csv", sep=",", engine='python')
+
+    mayo_mri_qc = pd.read_csv(
+        path.join(csv_dir, "MAYOADIRL_MRI_IMAGEQC_12_08_15.csv"),
```

Comment:
Since `csv_dir` is now a `pathlib.Path` instance:

```suggestion
        csv_dir / "MAYOADIRL_MRI_IMAGEQC_12_08_15.csv",
```